src/main/java/seedu/address/logic/commands/RemoveConfirmation.java

```diff
@@ -0,0 +1,37 @@
+package seedu.address.logic.commands;
+
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.ui.CommandBox;
+import seedu.address.logic.parser.RemoveCommandParser;
+
+/**
+ * Prompts user confirmation upon remove to ensure safe deletion.
+ */
+public abstract class RemoveConfirmation extends Command {
+    protected static final String MESSAGE_INVALID_DECISION_MAKING = "'yes' or 'no' is only used for decision making!"
+            + "Please start with a valid command word!";
+
+    private final RemoveCommandParser parserHelper = new RemoveCommandParser();
+
+    // TO PROPERLY IMPLEMENT!! - done, left with slap
+    // to check if previous command from command history is remove INDEX, as in other cases, typing in the command
+    // "yes" or "no" will not make sense
+    public boolean isValidConfirmation() {
```

Comment:
Perhaps this portion can be abstracted out a bit more? It is starting to look like an arrowhead

Reply:
Thanks for pointing out, will fix it!